OCPP-J/src/main/java/eu/chargetime/ocpp/WebSocketListener.java

```diff
@@ -161,6 +169,11 @@ public void onStart() {
 
     configure();
     server.start();
+    try {
+        socketBound.get(SOCKET_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    } catch (Exception e) {
+        throw new SocketException("Unable to bind server socket");
```

Comment:
suggestion: catch more specifically than `Exception`

suggestion: pass exception to new exception for better causal awareness (constructor requires using more generic `IOException`)

```suggestion
    } catch (TimeoutException e) {
        throw new IOException("Timeout while waiting for WebSocketServer to open", e);
```

question: Do we want to catch exceptions from `onError()`? These could presumably trigger prior to the future timeout and would be useful error information to propagate.

issue: Does this result in UB when it takes longer than `SOCKET_TIMEOUT_MILLIS` to connect? Couldn't this leave the server in a connected state but the calle having received an exception? Maybe we need to close the socket in this scenario

Reply:
Good catch on closing the server within the exception handler.

For catching exceptions from `onError()`, how do you recommend doing that?

Reply:
> For catching exceptions from onError(), how do you recommend doing that?

I think a `socketBound.completeExceptionally(ex)` at the end of `onError()` would work

Reply:
I had already implemented with a local method variable, but using `completeExceptionally` is more elegant.  Thanks.

Addressed your two suggestions, the question, and the issue